packages/manager/apps/ips/src/data/hooks/ip/useGetDedicatedServerTasks.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  useQueries,
+  useQueryClient,
+  UseQueryOptions,
+} from '@tanstack/react-query';
+import { ApiError, ApiResponse } from '@ovh-ux/manager-core-api';
+import {
+  DedicatedServerTaskResponse,
+  getDedicatedServerTask,
+  getDedicatedServerTaskQueryKey,
+  getDedicatedServerTasks,
+  getDedicatedServerTasksQueryKey,
+} from '@/data/api';
+import {
+  INVALIDATED_REFRESH_PERIOD,
+  UPDATE_TASKS_QUERY_KEY_PARAMS,
+} from '@/utils';
+
+export type UseGetDedicatedServerTasksParams = {
+  serviceName: string;
+  functionList: string[];
+  statusList: string[];
+  enabled: boolean;
+  queryKeyToInvalidate: string[];
+};
+
+function isUpdateTaskStatus(status: string): boolean {
+  return status
+    ? UPDATE_TASKS_QUERY_KEY_PARAMS.statusList.includes(status)
+    : true;
+}
+
+// Only for dedicated server for now
+export const useGetDedicatedServerTasks = ({
+  serviceName,
+  functionList,
+  statusList,
+  enabled = true,
+  queryKeyToInvalidate,
+}: UseGetDedicatedServerTasksParams) => {
+  const queryClient = useQueryClient();
+
+  const parameterSets: Array<{ taskFunction: string; status: string }> = [];
+
+  functionList.forEach((taskFunction) => {
+    statusList.forEach((status) => {
+      parameterSets.push({ taskFunction, status });
+    });
+  });
```

Comment:
use useMemo here, we don't want to compute this on every re-render in my opinion
Also avoid mutations

```
const parameterSets = React.useMemo(() => functionList.flatMap(taskFunction => statusList.map(status => ({taskFunction, status})), []);
```